### ISSUE TITLE
Investigate reconnecting sensor

### ISSUE THREAD

**Issue by markoaamunkajo** [ruuvi/com.ruuvi.station.ios#2148]
Description:

Investigate if we will find a way to reconnect sensor if it was disconnected

**Comment by priyonto** [comment]
Currently it _works_ usually as I have been testing. But, I have also seen it not working at times, specially when system kills the connections. Based on my investigation so far everything seems to be placed correctly to try reconnecting when the connections are removed whether by killing the app or otherwise. My guess it has something to do with number of sensors being connected, not sure if there are any shadow restrictions from system side to limit how many peripherals can be and stay connected to the device. I planned to give a couple of more days before moving the card to backlog.

**Comment by priyonto** [comment]
I tested with 5 sensors and did not kill the app from background or open the app to foreground for a week while using the device usually with bunch of apps from my regular use, as its my personal device. I had no issue in connection after a week with any of the sensors. Within this period I also went outside, so tags were disconnected, but then reconnected automatically when I came back. I did not investigate the background data yet, but since the connections are restored I am expecting data should be ideally collected in background as usual. 

So not really sure what's up with this. It could be an issue on older devices probably, or some other unknown issue. So, I am moving this and #2142 to backlog until next time.

**Comment by markoaamunkajo** [comment]
If sensor connected and app killed, reconnects on reopening app - yes
Intentionally disconnect tag and app killed, disconnected on reopening app - yes
Sensor connected and sensor card deleted and re-added, not connected by default - yes
Background scanning turned off when sensor was connected, re-enabling bg scanning will connect sensor - no, this swill leave sensor to spinning wheel and not connect sensor

**Comment by priyonto** [comment]
@markoaamunkajo 

> Background scanning turned off when sensor was connected, re-enabling bg scanning will connect sensor - no, this swill leave sensor to spinning wheel and not connect sensor

Currently we disconnect all connections on background scanning disabled. What do we do in that case? Fix the "spinning" to now spin all and let users connect them or keep a log of previously connected sensors and try to reconnect? This reconnection one is obviously tricky solution when user explicitly disabled background scanning.

**Comment by markoaamunkajo** [comment]
You are right @priyonto , maybe user intentionally disabled background scanning (maybe due to another iOS-device connecting) and we cannot know if user actually wants this.

Maybe this is not worth fixing.
Moving card to done.
